### DISCUSSION TITLE
Extending OpenSSHJunOSRIPing

### DISCUSSION THREAD

**Discussion by oetiker** [oetiker/SmokePing#309] [Q&A]
copied a missplaced post

I am looking to extend OpenSSHJunOSRIPing by adding routing-instance support to Probe. I tried adding $routing_instance variable in the pingone and targetvars subroutines but keep getting the unknown variable whenever I tried to add routing_instance to the probe config. Any thoughts?

```
sub pingone ($$){
...
    my $password = $target->{vars}{junospass};
    my $routing_instance = $target->{vars}{routing_instance};
    my $bytes = $self->{properties}{packetsize};
...

                junospass => {
                       _doc => <<DOC,
The junospass option allows you to specify the password for the username
specified with the option junosuser.
DOC
                       _example => 'password',
               },
               routing_instance => {
                       _doc => 'The routing-instance the ping should be sent from.',
                       _example => 'default',
                       _re => '.+',
               },
```

ERROR: /config/Probes, line 31: unknown variable 'routing_instance'

**Comment by oetiker** [comment]
my guess is that one of two things happend

a) the config file you are trying to load has an issue ... that new variable will only be available in a section with the JunosRIPing probe
b) smokeping is not actually loading your modified copy of the probe

**Comment by penguin02007** [comment]
I think is more a? if I hardcode stuff in the module, it works. 

But for some reason it can't find the Class variable, this is what I am trying - 

https://github.com/oetiker/SmokePing/compare/master...penguin02007:add_routing_instances?expand=1

**Comment by oetiker** [comment]
please show your config file ...

**Comment by penguin02007** [comment]
Please see below 
```
*** Probes ***

+ FPing

binary = /usr/bin/fping

+ FPing6

binary = /usr/bin/fping
protocol = 6

+ DNS
binary = /usr/bin/dig
lookup = google.com
pings = 5
step = 300

+OpenSSHJunOSPing

 forks = 280
 offset = 50%
 packetsize = 65
 step = 120
 timeout = 120
 tos = 0

 # The following variables can be overridden in each target section
 junospass = foo
 junosuser = bar
 pings = 5
 routing_instance = mgmt

+EchoPingHttp

 binary = /usr/bin/echoping
 forks = 5
 offset = 50%
 step = 300
# The following variables can be overridden in each target section
 accept_redirects = yes
 extraopts = -some-letter-the-author-did-not-think-of
 ignore_cache = yes
 ipversion = 4
 pings = 5
 port = 80
 priority = 6
 revalidate_data = no
 timeout = 20
 tos = 0xa0
 url = /
 waittime = 1
 ```

**Reply by oetiker** [reply]
ist it `OpenSSHJunOSPing` which you extended? not `OpenSSHJunOSRIPing` ?

**Reply by penguin02007** [reply]
Yes sorry, we extended `OpenSSHJunOS.pm` and renamed to `OpenSSHJunOSRIPing.pm`. The original module in question is OpenSSHJunOS.
